hyperhtml.js

```diff
@@ -278,15 +305,61 @@ var hyperHTML = (function () {'use strict';
   // Helpers
   // -------------------------
 
-  // it does exactly what it says
-  function appendNodes(node, childNodes) {
-    for (var
-      i = 0,
-      length = childNodes.length;
-      i < length; i++
-    ) {
-      node.appendChild(childNodes[i]);
+  // given a generic text or comment node
+  // remove the surrounding emptiness
+  function cleanAround(textNode, sibling) {
+    var adj = textNode[sibling];
+    if (adj && adj.nodeType === 3 && trim.call(adj.textContent).length < 1) {
+      textNode.parentNode.removeChild(adj);
+      cleanAround(textNode, sibling);
+    }
+  }
+
+  // find a live node through a virtual one
+  function find(live, virtual) {
+    var i, length, name, parentNode, map = [];
```

Comment:
in case you haven't noticed, there is never more than a single `var` declaration per function so nobody can ever possibly be confused by the hoisting nature of the `var`.

I'd rather go ES6 when the time is right. 'till that day, I'd like to have zero tooling dependencies, hoping whoever uses this lib is also not transpiling much more than just  what `babel-plugin-transform-es2015-template-literals` would do :wink: